src/Illuminate/Contracts/Routing/ControllerDispatcher.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Illuminate\Contracts\Routing;
+
+use Illuminate\Routing\Route;
+
+interface ControllerDispatcher
+{
+    /**
+     * Dispatch a request to a given controller and method.
+     *
+     * @param  Route  $route
+     * @param  mixed  $controller
+     * @param  string  $method
+     * @return mixed
+     */
+    public function dispatch(Route $route, $controller, $method);
```

Comment:
This is slightly unfortunate as typically a contract can't depend on any outside library. This depends on Illuminate\Routing.